cuqi/model/_model.py

```diff
@@ -160,334 +241,808 @@ def range_dim(self):
         """
         return self.range_geometry.par_dim
 
-    def _2fun(self, x, geometry, is_par):
-        """ Converts `x` to function values (if needed) using the appropriate 
-        geometry. For example, `x` can be the model input which need to be
-        converted to function value before being passed to 
-        :class:`~cuqi.model.Model` operators (e.g. _forward_func, _adjoint_func,
-        _gradient_func).
+    def _check_domain_geometry_consistent_with_forward(self):
+        """Private function that checks if the domain geometry of the model is
+        consistent with the forward operator."""
+
+        non_default_args = cuqi.utilities.get_non_default_args(self._forward_func)
+        if (
+            not isinstance(
+                self.domain_geometry, cuqi.experimental.geometry._ProductGeometry
+            )
+            and len(non_default_args) > 1
+        ):
+            raise ValueError(
+                "The forward operator input is specified by more than one argument. This is only supported for domain geometry of type tuple of cuqi.geometry.Geometry objects."
+            )
```

Comment:
This is rather besides the point of this PR, but the formatting here using (i think) black makes the review somewhat harder in my opinion.

Reply:
I know 🙂, I left it as is for now which is auto-formatted.